### DISCUSSION TITLE
How to get Ethereum Pairs?

### DISCUSSION THREAD

**Discussion by keith-johnson963** [toptal126/UniswapV2_Pair_Scanner#1] [Q&A] [answered]
Hi @toptal126 
I need to fetch all pairs from Uniswap v2 and v3 on Ethereum mainnet?
What should I do to so?

**Comment by realxtefi** [comment, marked as answer]
Hi @kadena-engineer .
First you need to get dedicated RPC Url for Ethereum mainnet (IHM, Infra.io has some limitation for daily calls number).
I suggest you to choose quicknode.
And replace the rpl url at the source code and list the Uniswap V2 Factory address.
V3 scanner is on development.
Thanks again
